VeganVirus/VeganVirus/ImageAction.cpp

```diff
@@ -0,0 +1,91 @@
+#include "ImageAction.h"
+
+//global var
+int num = 0;
+
+
+ImageAction::ImageAction(double req, Draw* draw) :
+    Action(req, POP_UP_ICON_PATH), _draw(draw)
+{
+    _time = 0;
+    _bmp = new Bitmap(L"popUp\\image0.png");
+}
+
+ImageAction::~ImageAction()
+{
+    delete _bmp;
+}
+
+/*
+move image to random location and set the time
+param: none
+return: none
+*/
+void ImageAction::start()
+{
+    //check that image don't start in out of bounce
+    _x = GetSystemMetrics(SM_CXSCREEN) - IMAGE_WIDTH;//rand() % (GetSystemMetrics(SM_CXSCREEN) - IMAGE_WIDTH);
+    _y = GetSystemMetrics(SM_CYSCREEN) - IMAGE_HIGH;// rand() % (GetSystemMetrics(SM_CYSCREEN) - IMAGE_HIGH - 1);
+
+    _time = POP_UP_DURATION;
+}
+
+/*
+the function check if to activate
+param dt: difference of time between each frame
+return: none
+*/
+void ImageAction::update(double dt)
+{
+    if (_time <= 0)
+    {
+        return;
+    }
+    _time -= dt;    
+    
+    //move(pixelPerSecond)
+    _x += movmentX * dt * 15;
+    _y += movmentY * dt * 15; 
```

Comment:
Magic number (15), which I assume is the movement speed.
I suggest adding a `#define MOVEMENT_SPEED 150` and setting the initial `movementX` and `movementY` to one. Or alternatively, increase the initial values of `movementX`, `movementY` to 150.